src/frontend/tsx/GamePanel.tsx

```diff
@@ -24,46 +28,66 @@ export function GamePanel () {
+      const move: number = (wIndex / 2) + 1
+      const bIndex: number = wIndex + 1
+      const wMove: SAN = displayMoves[wIndex]
+      const bMove: SAN = displayMoves[wIndex + 1]
+
+      if (bIndex > displayMoves.length) {
+        components.push(
+          <li key={ move } className='move-item' style={{ opacity: moveOpacity(wIndex) }}>
+            <span onClick={ () => setDisplayIndex(wIndex) }>
+              { wMove }
+            </span>
+          </li>
+        )
+      } else {
+        components.push(
+          <li key={ move } className='move-item' style={{ opacity: moveOpacity(wIndex) }}>
+            <span onClick={ () => setDisplayIndex(wIndex) }>
+              { wMove }
+            </span>
+            { '\xa0'.repeat(6 - wMove.length) }
+            <span onClick={ () => setDisplayIndex(bIndex) } style={{ opacity: (wIndex <= displayIndex && bIndex > displayIndex) ? 0.3 : 1.0 }}>
+              { bMove }
+            </span>
+          </li>
+        )
+      }
+    }
+
+    return components
+  }
+
   return (
     <div className='game-panel-container col'>
       <div className="game-panel col">
-        <div id="opp-timer" className={'timer row' + (hasGame ? '' : ' invisible')}>
+        <div id="opp-timer" className={'timer row' + (hasGame ? '' : ' hidden')}>
           <p>00:00</p>
         </div>
-        <div id="opp-player" className={'player row' + (hasGame ? '' : ' invisible')}>
-          <p>~sampel-palnet</p>
+        <div id="opp-player" className={'player row' + (hasGame ? '' : ' hidden')}>
+          <p>{opponent}</p>
         </div>
-        <div className="moves col">
-          <div className="moves-divider"></div>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
+        <div className={'moves col' + (hasGame ? '' : ' hidden')}>
+          <ol>
+            { moveList() }
+          </ol>
         </div>
-        <div id="our-player" className={'player row' + (hasGame ? '' : ' invisible')}>
-          <p>~sampel-palnet</p>
+        <div id="our-player" className={'player row' + (hasGame ? '' : ' hidden')}>
+          <p>~{window.ship}</p>
         </div>
-        <div id="our-timer" className={'timer row' + (hasGame ? '' : ' invisible')}>
+        <div id="our-timer" className={'timer row' + (hasGame ? '' : ' hidden')}>
```

Comment:
We should probably make a separate "Practice Panel" that contains the "Reset Practice Board" button and can do it's own thing. For now, it would just have that button in it.

When a game against a human opponent is active, the entire "Practice Panel" would be hidden, and the "Game Panel" would be visible. Otherwise, it would be vice versa. The old, old version of the chess frontend used to do this. I think the code is as simple as:
```
if (activeGame) {
  <GamePanel />
} else {
  <PracticePanel />
}
```